src/mame/drivers/schick.cpp

```diff
@@ -0,0 +1,611 @@
+		if (src & 0x80) j = 7 - j;
+
+		/* decode the ROM data */
+		rom[A] = src ^ data_xortable[i][j];
+
+		/* now decode the opcodes */
+		/* pick the translation table from bits 4, 8 and 12 of the address */
+		i = ((A >> 4) & 1) + (((A >> 8) & 1) << 1) + (((A >> 12) & 1) << 2);
+		m_decrypted_opcodes[A] = src ^ opcode_xortable[i][j];
+	}
+
+	for (int A = end; A < nodecend; A++)
+	{
+		m_decrypted_opcodes[A] = rom[A];
+	}
+}
+
+void schick_state::decode_schick_extra(int size, uint8_t* rom)
+{
+	// schick has an extra layer of encryption in addition to the penta encryption
+	for (int A = 0x0000; A < 0x8000; A++)
+	{
+		uint8_t srcdec = rom[A];
+
+		if (A & 0x100)
+		{
+			srcdec = bitswap<8>(srcdec^0x41, 7, 4, 5, 6, 3, 2, 1, 0);
+		}
+		else
+		{
+			srcdec = bitswap<8>(srcdec^0x51, 7, 6, 5, 0, 3, 2, 1, 4);
+		}
+
+
+		rom[A] = srcdec;
+	}
+
+	for (int A = 0x8000; A < 0x10000; A++) // TODO: verify everything
+	{
+		uint8_t srcdec = rom[A];
+
+		// this layer of encryption only affects bits 0,4,6 ?
+
+		if (rom == m_decrypted_opcodes)
+		{
+			if (A & 0x1000) // TODO: more conditions?
+			{
+				srcdec = bitswap<8>(srcdec ^ 0x40, 7, 0, 5, 6, 3, 2, 1, 4);
+				rom[A] = srcdec;
+			}
+			else
+			{
+				if  (A & 0x100)
+				{
+					srcdec = bitswap<8>(srcdec ^ 0x40, 7, 4, 5, 0, 3, 2, 1, 6);
+					rom[A] = srcdec;
+				}
+				else
+				{
+					srcdec = bitswap<8>(srcdec ^ 0x41, 7, 0, 5, 6, 3, 2, 1, 4);
+					rom[A] = srcdec;
+				}
+			}
+		}
+		else
+		{
+			// // TODO: more conditions?
+			if (A & 0x10)
+				srcdec = bitswap<8>(srcdec ^ 0x11, 7, 0, 5, 6, 3, 2, 1, 4);
+			else
+				srcdec = bitswap<8>(srcdec ^ 0x51, 7, 4, 5, 0, 3, 2, 1, 6);
+
+			rom[A] = srcdec;
+		}
+	}
+}
+
+void schick_state::init_schick()
+{
+	uint8_t *rom = memregion("maincpu")->base();
+
+	decode_penta(0x8000, 0x10000);
+
+	decode_schick_extra(0x10000, rom);
+	decode_schick_extra(0x10000, m_decrypted_opcodes);
+}
+
+
+// MH032288 PCB. LC ('lato componenti', components side in Italian) so maybe produced in Italy?
+// A plastic block in a corner covers probably the main CPU and the decryption logic.
+// Might fit better in a different driver once decrypted
+ROM_START( schick )
+	ROM_REGION( 0x10000, "maincpu", 0 )
+	ROM_LOAD( "27c512.8d", 0x0000, 0x10000, CRC(38986329) SHA1(bfd62d6a49d25acc582e5f076833c3b22c1a7fd7) )
+
+	ROM_REGION( 0x8000, "audiocpu", 0 )
+	ROM_LOAD( "27c256.7m", 0x0000, 0x8000, CRC(694dadca) SHA1(65d436d6c8ebf6a9b5af286122e7391973d463e0) ) // identical to bombjack, but 4x
+
+	ROM_REGION( 0x10000, "gfx1", 0 )
+	ROM_LOAD( "4.27c256.4f",  0x0000, 0x8000, CRC(f666a52a) SHA1(877e1112c9c9a39b55934f6a382ad35fc9bf6859) ) // only labeled ROM
+	ROM_LOAD( "27c256.4e",    0x8000, 0x8000, CRC(edb4a243) SHA1(64f35b5142ffb3bfbd6a2899af9d1d719b83e2a1) )
+
+	ROM_REGION( 0x0100, "proms", ROMREGION_ERASEFF )
+	ROM_LOAD( "proms",    0x0000, 0x0100, NO_DUMP ) // color palette, unknown PROM size + arrangement
```

Comment:
FWIW the dumper wrote 'I see two PROMs on the board, an MMI 63S081N and an AMD AM27S33PC'

Though they could be for colors or for something else.

Thanks for looking at this, BTW

Thanks for looking at this, BTW

Reply:
if you want to add them with the correct sizes as NO_DUMP that's fine, clearly the pengo palette decode isn't going to be correct even with the PROMs tho as these are 4bpp tiles, not 2bpp tiles

Reply:
also if you want to add yourself to the header, I have no complaints either

Reply:
they definitely added a bunch of protections anyway
[:maincpu] ':maincpu' (E3F5): unmapped io memory read from 000A & FF

this read happens, and if the code gets executed after it, then you end up getting tons of bonus lives, again I guess it should never be taken...

Reply:
service mode ROM fail is probably just because it's a hack, I think the decryption is correct now, just it's littered with all these extra checks.